File: src/main/java/dev/kensa/output/MultiFileWriter.java
package dev.kensa.output;

import dev.kensa.Kensa;
import dev.kensa.context.TestContainer;
import dev.kensa.output.template.Template;

import java.util.Set;
import java.util.function.BiConsumer;

import static dev.kensa.output.template.Template.Mode.MultiFile;
import static dev.kensa.output.template.Template.Mode.TestFile;
import static dev.kensa.output.template.Template.asIndex;
import static dev.kensa.output.template.Template.asJsonScript;

public class MultiFileWriter implements BiConsumer<Set<TestContainer>, Kensa.Configuration> {
    @Override
    public void accept(Set<TestContainer> containers, Kensa.Configuration configuration) {
        Template indexTemplate = configuration.createTemplate("index.html", MultiFile);

        for (TestContainer container : containers) {
            Template scriptTemplate = configuration.createTemplate(container.testClass().getName() + ".html", TestFile);
            scriptTemplate.addJsonScript(container, asJsonScript(configuration.renderers()));
            scriptTemplate.write();
            indexTemplate.addIndex(container, asIndex());
        }

        indexTemplate.write();
    }
}

File: src/main/java/dev/kensa/output/MultiFileWriter.kt
package dev.kensa.output

import dev.kensa.Configuration
import dev.kensa.context.TestContainer
import dev.kensa.output.template.Template.Companion.asIndex
import dev.kensa.output.template.Template.Companion.asJsonScript
import dev.kensa.output.template.Template.Mode.MultiFile
import dev.kensa.output.template.Template.Mode.TestFile

class MultiFileWriter : (Set<TestContainer>, Configuration) -> Unit {
    override fun invoke(containers: Set<TestContainer>, configuration: Configuration) {
        configuration.createTemplate("index.html", MultiFile).also { index ->
            containers.forEach { container ->
                configuration.createTemplate(container.testClass.qualifiedName + ".html", TestFile).also { script ->
                    script.addJsonScript(container, asJsonScript(configuration.renderers))
                    script.write()
                }
                index.addIndex(container, asIndex())
            }
            index.write()
        }
    }
}